app/models/casa_case.rb

```diff
@@ -54,11 +54,30 @@ def has_transitioned?
   end
 
   def update_cleaning_contact_types(args)
+    begin
+      args = parse_court_date(args)
+    rescue Date::Error
+      errors.messages[:court_date] << "was not a valid date."
+      return false
+    end
     transaction do
       casa_case_contact_types.destroy_all
       update(args)
     end
   end
+
+  private
+
+  def parse_court_date(args)
+    day = args.delete("court_date(3i)")
+    month = args.delete("court_date(2i)")
+    year = args.delete("court_date(1i)")
+    return args if day.blank? && month.blank? && year.blank?
+    raise Date::Error if day.blank? || month.blank? || year.blank?
+    court_date = Date.parse("#{day}-#{month}-#{year}")
+    args["court_date"] = court_date
+    args
+  end
```

Comment:
Rails is smart enough to put the 3 separate pieces of the date together into the date field -- can we simplify this a bit by just returning an error if `court_date(1i)`, `court_date(2i)`, or `court_date(3i)` is blank?